tests/Basics/SupervisedBufferTest.cpp

```diff
@@ -0,0 +1,479 @@
+      ASSERT_GE(monitor.current(), smallValue.memoryUsage());
+    }
+    ASSERT_EQ(monitor.current(), smallValue.memoryUsage());
+    monitor.decreaseMemoryUsage(smallValue.memoryUsage());
+    smallValue.destroy();
+    ASSERT_EQ(monitor.current(), 0);
+  }
+}
+
+TEST(SupervisedbufferTest,
+     ManuallyIncreaseAccountsMemoryLargeAndSmallValuesSupervisedBuffer) {
+  auto& global = GlobalResourceMonitor::instance();
+  ResourceMonitor monitor{global};
+
+  {
+    ResourceUsageScope usageScope(monitor);
+    AqlValue largeValue;
+    std::size_t sizeBeforeLocal = 0;
+    std::size_t valueSizeLocal = 0;
+    {
+      SupervisedBuffer supervisedBuffer(monitor);
+      Builder builder(supervisedBuffer);
+      builder.openArray();
+      builder.add(Value(std::string(2048, 'a')));
+      builder.close();
+      ASSERT_GT(monitor.current(), 0);
+
+      std::size_t monitorBefore = monitor.current();
+      std::size_t sizeBefore = builder.size();
+      usageScope.increase(sizeBefore);
+      ASSERT_EQ(monitor.current(), monitorBefore + sizeBefore);
+
+      sizeBeforeLocal = sizeBefore;
+      largeValue = AqlValue{builder.slice(), builder.size()};
+      valueSizeLocal = largeValue.memoryUsage();
+      // account the value's memory on top of manual buffer accounting
+      monitor.increaseMemoryUsage(valueSizeLocal);
+      ASSERT_GE(monitor.current(), sizeBefore + largeValue.memoryUsage());
+    }
+    ASSERT_GE(monitor.current(), sizeBeforeLocal + valueSizeLocal);
+    // drop our manual accounting for the value before destroying it
+    monitor.decreaseMemoryUsage(valueSizeLocal);
+    largeValue.destroy();
+    usageScope.decrease(sizeBeforeLocal);
+    ASSERT_EQ(monitor.current(), 0);
+  }
+
+  {
+    ResourceUsageScope usageScope(monitor);
+    AqlValue smallValue;
+    std::size_t sizeBeforeLocal = 0;
+    std::size_t valueSizeLocal = 0;
+    {
+      SupervisedBuffer supervisedBuffer(monitor);
+      Builder builder(supervisedBuffer);
+      builder.openArray();
+      builder.add(Value(42));
+      builder.close();
+      ASSERT_GT(monitor.current(), 0);
+
+      std::size_t monitorBefore = monitor.current();
+      std::size_t sizeBefore = builder.size();
+      usageScope.increase(sizeBefore);
+      ASSERT_EQ(monitor.current(), monitorBefore + sizeBefore);
+
+      sizeBeforeLocal = sizeBefore;
+      smallValue = AqlValue{builder.slice(), builder.size()};
+      valueSizeLocal = smallValue.memoryUsage();
+      monitor.increaseMemoryUsage(valueSizeLocal);
+      ASSERT_GE(monitor.current(), sizeBefore + smallValue.memoryUsage());
+    }
+    ASSERT_GE(monitor.current(), sizeBeforeLocal + valueSizeLocal);
+    monitor.decreaseMemoryUsage(valueSizeLocal);
+    smallValue.destroy();
+    usageScope.decrease(sizeBeforeLocal);
+    ASSERT_EQ(monitor.current(), 0);
+  }
+}
+
+TEST(SupervisedbufferTest,
+     ManuallyIncreaseAccountsMemoryLargeAndSmallValuesNormalBuffer) {
+  auto& global = GlobalResourceMonitor::instance();
```

Comment:
I do not understand the benefit of this test, it tests nothing on SupervisedBuffer

Reply:
just to know we haven't broken anything, but I can remove it